src/sql/src/plan/scope.rs

```diff
@@ -198,19 +199,23 @@ impl Scope {
     pub fn all_items<'a>(
         &'a self,
         outer_scopes: &'a [Scope],
-    ) -> impl Iterator<Item = (ColumnRef, &ScopeItem)> + 'a {
+    ) -> impl Iterator<Item = (ColumnRef, usize, &ScopeItem)> + 'a {
         // These are in order of preference eg
         // given scopes A(B(C))
         // items from C should be preferred to items from B to items from A
+        let mut lat_level = 0;
```

Comment:
IIUIC `lat_level` is basically a way of grouping adjacent scopes that define a name space for ambiguity detection purposes. It only started making sense to me after looking at the test. Perhaps we could document this method containing one of the examples in the test, because they really help.